src/main.rs

```diff
@@ -32,32 +35,61 @@ impl Terminal {
 
     fn should_ask_for_todo(&mut self) -> bool {
         let mut buf = String::new();
-        self.stdin.read_line(&mut buf).unwrap();
+
+        if let Err(_) = self.stdin.read_line(&mut buf) {
+            self.show_error(Err(TerminalError::StdoutErr));
+        }
 
         buf.trim() == "s"
     }
 
     fn ask_for_new_todo(&mut self) -> Todo {          
 
         println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
-        
-        let response = self.should_ask_for_todo();
-        
-        if !response {
+                
+        if !self.should_ask_for_todo() {
             println!("\nTodo list finalizado! 🤠");
             std::process::exit(0);            
         } 
 
         println!("\nQual TODO deseja criar?");
 
         let mut new_todo = String::new();
-        self.stdin.read_line(&mut new_todo).unwrap();
+
+        if let Err(_) = self.stdin.read_line(&mut new_todo) {
+            self.show_error(Err(TerminalError::StdoutErr));
+        }
+
         let todo_message = new_todo.trim().to_string();
 
         Todo { message: todo_message }
     }
 
-    fn show_todo(&mut self, todo: &Todo) {        
-        writeln!(self.stdout, "\n✅: {}", todo.message).unwrap();
+    fn show_todo(&mut self, todo: &Todo) -> Result<(), io::Error> { 
+        writeln!(self.stdout, "\n✅: {}", todo.message)?;
+        Ok(())
+    }
+
+    fn show_error(&mut self, data: Result<(), TerminalError>) {
+
+        if let Err(error) = data {
+            println!("{}", TerminalError::message_err(error));
+        }
+       
+    }
```

Comment:
Como a função `show_error` _sempre_ será usada para imprimir um erro na tela, aceitar um `Result` como parâmetro me parece desnecessário, WDYT?

```suggestion
    fn show_error(&mut self, error: TerminalError) {
            println!("{}", error.message_err());
    }
```

A vantagem de mudar a assinatura é que:
1. Você simplifica a implementação (afinal, sempre terá algo que representa um erro), de modo a poder remover o `if`.
2. Garante que só será possível chamar a função se você tiver um `TerminalError`, isto é, não poderá, por exemplo, passar um `Ok(())`.

Reply:
Com certeza! Será implementado assim.